### ISSUE TITLE
Add a setting to toggle the “search as you type” behaviour

### ISSUE THREAD

**Issue by knowler** [npmx-dev/npmx.dev#1710]
As a basis for improving the accessibility of the default search experience, we need to add a setting for toggling the _search as you type_ behaviour (default can be toggled _on_). This should make the search field and search button behave how they normally would: nothing should change until the user presses <kbd>Enter</kbd>. This also means the behaviour of going to the first result when pressing <kbd>Enter</kbd> should be affected by the toggle.

**Comment by alex-key** [comment]
Should we put it under "DATA SOURCE" section in Settings and have smth like this?

<img width="1195" height="579" alt="Image" src="https://github.com/user-attachments/assets/f12caa43-a84f-4e6c-a381-77f93279afad" />

Current:

<img width="1193" height="446" alt="Image" src="https://github.com/user-attachments/assets/ddd1a120-ebff-4373-a1aa-f4005c4b8353" />

**Comment by knowler** [comment]
@alex-key Yes, that’s exactly where I imagined it would be. We can always workshop the name and description in the PR. We could maybe call the setting “Instant search” and then describe that it “navigates to the search results as you type and opens the first result on <kbd>Enter</kbd>” or something like that.

**Comment by alex-key** [comment]
Cool, will do this
